### PR TITLE
upgrade to codemirror v19

### PR DESCRIPTION
**Pull request by Nexucis**
Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

This PR is going to upgrade codemirror to the last version. There are some breaking changes in lezer and it looks like the behavior of the parser changed.
Some tests in the autocompletion failed and need to be fixed before thinking to merge this PR.

fix prometheus-community/codemirror-promql#150
fix prometheus-community/codemirror-promql#149

### CODE REVIEW

**Comment by Nexucis** [comment]
this PR once it is finished will fix https://github.com/prometheus-community/codemirror-promql/issues/150 and fix https://github.com/prometheus-community/codemirror-promql/issues/149

**Comment by Nexucis** [comment]
and don't know if there is a link, but upgrading codemirror lead to have a `node_modules` folder in the `react-app` which is not the case currently on the `main` branch

**Comment by juliusv** [comment]
Looks good so far. Weird that it would create a `node_modules` dir that wasn't there before, would be good to find out why.

**Comment by Nexucis** [comment]
should be good for a review now @juliusv I fix the code of the autocompletion.

The behavior changed for the expression like `http_requests_total{method="GET"} off`. Before it was identified as a Binary expression. Now it's just a correct Expression + an error node.

For the new `node_modules` folder that appears in the `react-app`, I think it's coming from a conflict on the babel deps. Don't know where is coming from so far ... That's just my best supposition of why it appeared.

File: web/ui/module/codemirror-promql/src/complete/hybrid.ts [hunk 1]
```diff
@@ -267,16 +267,29 @@ export function analyzeCompletion(state: EditorState, node: SyntaxNode): Context
...
+          // this last case can appear with the following expression:
+          // 1. http_requests_total{method="GET"} off
+          // 2. rate(foo[5m]) un
+          // 3. sum(http_requests_total{method="GET"} off)
+          // For these different cases we have this kind of tree:
+          // Parent (
+          //    Expr(),
+          //    ⚠(Identifier)
+          // )
+          // We don't really care about the parent, here we are more interesting if In the siblings of the error node, there is the node 'Expr'
```
Comment (juliusv) [inline web/ui/module/codemirror-promql/src/complete/hybrid.ts:281]:
```suggestion
          // We don't really care about the parent, here we are more interested if in the siblings of the error node, there is the node 'Expr'
```

File: web/ui/module/codemirror-promql/src/complete/hybrid.ts [hunk 2]
```diff
@@ -267,16 +267,29 @@ export function analyzeCompletion(state: EditorState, node: SyntaxNode): Context
...
+          // If it is the case, then likely we should autocomplete or the BinOp or the offset.
```
Comment (juliusv) [inline web/ui/module/codemirror-promql/src/complete/hybrid.ts:282]:
```suggestion
          // If it is the case, then likely we should autocomplete the BinOp or the offset.
```

**Comment by juliusv** [comment]
:+1: besides the comments

**Pull request merged**